tpm2/error.go

```diff
@@ -0,0 +1,325 @@
+)
+
+var fmt0Msg = map[RcFmt0]string{
+	RcInitialize:      "TPM not initialized by TPM2_Startup or already initialized",
+	RcFailure:         "commands not being accepted because of a TPM failure",
+	RcSequence:        "improper use of a sequence handle",
+	RcPrivate:         "not currently used",
+	RcHMAC:            "not currently used",
+	RcDisabled:        "the command is disabled",
+	RcExclusive:       "command failed because audit sequence required exclusivity",
+	RcAuthType:        "authorization handle is not correct for command",
+	RcAuthMissing:     "5 command requires an authorization session for handle and it is not present.",
+	RcPolicy:          "policy failure in math operation or an invalid authPolicy value",
+	RcPCR:             "PCR check fail",
+	RcPCRChanged:      "PCR have changed since checked",
+	RcUpgrade:         "TPM is in field upgrade mode unless called via TPM2_FieldUpgradeData(), then it is not in field upgrade mode",
+	RcTooManyContexts: "context ID counter is at maximum",
+	RcAuthUnavailable: "authValue or authPolicy is not available for selected entity",
+	RcReboot:          "a _TPM_Init and Startup(CLEAR) is required before the TPM can resume operation",
+	RcUnbalanced:      "the protection algorithms (hash and symmetric) are not reasonably balanced. The digest size of the hash must be larger than the key size of the symmetric algorithm.",
+	RcCommandSize:     "command commandSize value is inconsistent with contents of the command buffer; either the size is not the same as the octets loaded by the hardware interface layer or the value is not large enough to hold a command header",
+	RcCommandCode:     "command code not supported",
+	RcAuthSize:        "the value of authorizationSize is out of range or the number of octets in the Authorization Area is greater than required",
+	RcAuthContext:     "use of an authorization session with a context command or another command that cannot have an authorization session",
+	RcNVRange:         "NV offset+size is out of range",
+	RcNVSize:          "Requested allocation size is larger than allowed",
+	RcNVLocked:        "NV access locked",
+	RcNVAuthorization: "NV access authorization fails in command actions",
+	RcNVUninitialized: "an NV Index is used before being initialized or the state saved by TPM2_Shutdown(STATE) could not be restored",
+	RcNVSpace:         "insufficient space for NV allocation",
+	RcNVDefined:       "NV Index or persistent object already defined",
+	RcBadContext:      "context in TPM2_ContextLoad() is not valid",
+	RcCPHash:          "cpHash value already set or not correct for use",
+	RcParent:          "handle for parent is not a valid parent",
+	RcNeedsTest:       "some function needs testing",
+	RcNoResult:        "returned when an internal function cannot process a request due to an unspecified problem. This code is usually related to invalid parameters that are not properly filtered by the input unmarshaling code",
```

Comment:
NIT: This has multiple sentences like where my last comment is, but ends with no period as it probably should?

Reply:
Normally I'd agree, but these are straight out of the TPM2 spec, Part 2, Table 16. I've seen other implementation who also use the exact strings for error messages. I can change these for consistent punctuation if you think it's better that way.

Reply:
don't use periods in error messages, in case they get wrapped with fmt.Errorf https://github.com/golang/go/wiki/CodeReviewComments#initialisms

Reply:
done